### PR TITLE
Fix OSX native dependency installation

### PR DESCRIPTION
**Pull request by hoyosjs**
- Avoid upgrading packages that are installed on the machine unless explicitly listed.
- Use a brewfile.

### CODE REVIEW

**Comment by hoyosjs** [comment]
@wtgodbe this fixes the internal build breaks

@jeffschwMSFT this fixes the official build (unrelated to the one you pinged earlier)

@akoeplinger do you mind reading over this? You are more familiar on this than most of us :)

File: eng/install-native-dependencies.sh [hunk 1]
```diff
@@ -14,7 +14,9 @@ elif [ "$1" = "OSX" ]; then
     if [ "$?" != "0" ]; then
         exit 1;
     fi
-    brew install icu4c openssl
+
+    engdir=$(dirname "${BASH_SOURCE[0]}")
```
Comment (akoeplinger) [inline eng/install-native-dependencies.sh:18]: `BASH_SOURCE` is only available in bash shells which is no longer the default on newer macOS versions. This script uses `#!/usr/bin/env sh` as the shebang so getting bash isn't guaranteed.

It'd be better to change the shebang to `#!/usr/bin/env bash` like we have in dotnet/runtime.

**Review by akoeplinger — APPROVED** [review]
Looks good with one suggestion.

**Review by jeffschwMSFT — APPROVED** [review]
Approved.  Please merge to unblock the build

**Pull request merged**